presto-cli/src/main/java/com/facebook/presto/cli/Query.java

```diff
@@ -231,8 +231,9 @@ private void pageOutput(OutputFormat format, List<String> fieldNames)
                 // ignore the user pressing ctrl-C while in the pager
                 ignoreUserInterrupt.set(true);
                 pager.getFinishFuture().thenRun(() -> {
-                    userAbortedQuery.set(true);
                     ignoreUserInterrupt.set(false);
+                    userAbortedQuery.set(true);
+                    client.close();
```

Comment:
We already do this in `renderResults()` in response to `QueryAbortedException`. Why does that not work?

Reply:
Ah, it could be that the delete fails because we interrupted the thread.